### ISSUE TITLE
Issue of suspiciously slow inference

### ISSUE THREAD

**Issue by Kaktusava** [lyclyc52/NeRF_RPN#20]
Hi authors,
Thank you for the great work! I've run into the problem that there's no way I can get the same inference speed as mentioned in the paper. The paper states 115ms, however I have tried measuring the time on the RTX 3060 using different methods such as [the benchmark method in run_rpn.py](github.com/lyclyc52/NeRF_RPN/blob/main/nerf_rpn/run_rpn.py#L595) and [the benchmark method in run_fcos.py](https://github.com/lyclyc52/NeRF_RPN/blob/main/nerf_rpn/run_fcos.py#L538). I also measured the forward pass time during evaluation in the script run_fcos.py. I got values of 3 seconds, 5.6 seconds and 8 seconds respectively. Can you please advise me what is my measurement mistake, or under what conditions was the time measured as mentioned in the paper?

**Comment by zymk9** [comment]
Hi. One thing we didn't make crystal clear in the paper is what exactly is the "inference time without postprocessing". We considered the bounding box decoding and proposal filtering as postprocessing, which is located here: https://github.com/lyclyc52/NeRF_RPN/blob/2f1d4ce6b2544433fb3e6ce9cf829bac587689cd/nerf_rpn/model/rpn.py#L507-L512

To reproduce the inference time, you may simply comment the lines above out. We rerun the benchmark on an RTX 3090 using the pretrained weights (anchor based, ResNet 50), and the result is 80 ms. I believe the data in the paper is from a 1080ti running a similar model.

Sorry for the confusion. Hope this helps.

**Comment by Kaktusava** [comment]
Thank you for your response!
Indeed, it became much faster, the benchmark anchor based, ResNet 50 shows _Average inference time: 206.5343 ms, std: 0.7214 ms._  on RTX3060.
Also for fcos model I found out that the main time is spent on nmscalculations. 
https://github.com/lyclyc52/NeRF_RPN/blob/2f1d4ce6b2544433fb3e6ce9cf829bac587689cd/nerf_rpn/model/fcos/utils.py#L109-L124

**Comment by zymk9** [comment]
Yes, we regard NMS as part of the post-processing and exclude it in the timing.
